File: firmware/mic_capture.py
"""Mic capture — grab audio chunks and stream them to the backend.

The backend feeds this audio to Deepgram STT. Firmware does NOT transcribe.

TODO: implement. This is a scaffold stub.
"""


def start_streaming(to: str) -> None:
    """Capture mic audio and send chunks to the backend.

    Args:
        to: backend audio endpoint (WebSocket/HTTP) from config.

    TODO:
      - open the mic (sounddevice)
      - read audio chunks at a suitable rate/format for Deepgram STT
      - stream chunks to `to` (websockets); reconnect on drop
    """
    raise NotImplementedError("TODO: mic capture + stream to backend")

File: firmware/src/mic_capture.cpp
// mic_capture.cpp — grab mic audio (QNX, C++).
//
// The laptop backend feeds this audio to Deepgram STT. Firmware does NOT
// transcribe — it just captures and ships chunks over the LAN.
//
// TODO: implement. This is a scaffold stub.

#include "firmware.hpp"

bool mic_open() {
    // TODO: open audio capture (QNX audio).
    return false;
}

bool mic_next_chunk(AudioChunk& out) {
    // TODO: read an audio chunk at a rate/format suitable for Deepgram STT.
    (void)out;
    return false;
}
